### PR TITLE
GDB-8060 support gdb10

### PR DESCRIPTION
**Pull request by svilenvelikov**
## What
Support GDB10.

## Why
There are clients which already use GDB10 via the library.

## How
Migrate all request handlers to account the changes in the GDB 10 API.
Refactor and update the tests.

### CODE REVIEW

**Comment by nikolaykolev** [comment]
Jenkins retry build

**Comment by nikolaykolev** [comment]
Jenkins retry build

File: test-e2e/tests/utils.js [hunk 1]
```diff
@@ -53,10 +54,30 @@ function createRepo(path) {
...
 function deleteRepo(name) {
   return axios({
     method: 'delete',
     url: `${Config.serverAddress}/rest/repositories/${name}`
+  }).catch((e) => {
+    throw new Error(e);
+  });
+}
+
+function deleteRepoSecurely(name) {
```
Comment (yordanalexandrov) [inline test-e2e/tests/utils.js:73]: This name seems strange, given the next line toggles security off

**Review by yordanalexandrov — DISMISSED** [review]

**Comment by svilenvelikov** [comment]
Jenkins retry build

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@
 * `Check the GDB server version and report mismatches if the client version implements a different version unless the check is stopped.`
 * `Support secured endpoints exposing only HTTPS`
 
+## 2.1.0 (2023-03-23)
```
Comment (teodossidossev) [inline CHANGELOG.md:9]: IMO we have to increment the major version as this version supports GDB 10 only

Comment (svilenvelikov) [reply CHANGELOG.md:9]: yeah, I changed it to 3.0.0

**Comment by yaskoo** [comment]
Jenkins retry build

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Comment by yaskoo** [comment]
Jenkins retry build

**Comment by yaskoo** [comment]
Jenkins retry build

**Comment by yaskoo** [comment]
Jenkins retry build

**Comment by yaskoo** [comment]
Jenkins retry build

**Comment by yaskoo** [comment]
Jenkins retry build

**Pull request merged**